### ISSUE TITLE
Proposal: add an ignoreRegexps option to max-len

### ISSUE THREAD

**Issue by Wilfred** [eslint/eslint#7343]
**What rule do you want to change?**

`max-len`

**Does this change cause the rule to produce more or fewer warnings?**

It adds an option so that `max-len` reports fewer warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Add an option `ignoreRegexps`, similar to the existing options `ignoreStrings` and `ignoreTemplateLiterals`. This would enable users to ignore long regexp literals with `max-len`.

**Please provide some example code that this change will affect:**

```js
/*eslint max-len: ["error", { "ignoreRegexps": true }]*/
var foo = /very very very very very very very very very very very very very very long/;
```

**What does the rule currently do for this code?**

Currently it reports: `Line 2 exceeds the maximum line length of 80. (max-len)`

**What will the rule do after it's changed?**

It will not warn.

**Comment by platinumazure** [comment]
Hi @Wilfred, thanks for the issue.

I think I understand what you're getting at, but just to be safe, could you please edit your post to conform with our [Rule Change Proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md)? Thanks!

**Comment by eslintbot** [comment]
Hi @Wilfred, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by Wilfred** [comment]
Sorry about that. I've updated the post.

**Comment by platinumazure** [comment]
Ack-- I do apologize, we have an existing issue for this: #3229.

Closing this as a duplicate.
